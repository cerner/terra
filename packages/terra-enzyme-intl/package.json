{
  "name": "@cerner/terra-enzyme-intl",
  "version": "4.6.1",
  "description": "Enzyme helpers for fortifying tests that depend on react-intl by decoupling the need for actual translations.",
  "author": "Cerner Corporation",
  "main": "lib/index.js",
  "engines": {
    "node": ">=10"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/cerner/terra.git",
    "directory": "packages/terra-enzyme-intl"
  },
  "bugs": {
    "url": "https://github.com/cerner/terra/issues"
  },
  "homepage": "https://github.com/cerner/terra/packages/terra-enzyme-intl#readme",
  "publishConfig": {
    "access": "public"
  },
  "license": "Apache-2.0",
  "keywords": [
    "Cerner",
    "enzyme",
    "i18n",
    "intl",
    "jest",
    "react",
    "react-intl",
    "Terra",
    "terra-enzyme-intl"
  ],
  "files": [
    "lib"
  ],
  "dependencies": {
    "prop-types": "^15.5.8"
  },
  "devDependencies": {
    "react-intl": "^5.25.1"
  },
  "peerDependencies": {
    "enzyme": "3.x",
    "react": "16.x",
    "react-intl": "^5.25.1"
  },
  "scripts": {
    "clean:install": "npm run clean && npm install",
    "compile": "babel --root-mode upward src --out-dir lib --copy-files",
    "jest": "jest --config ../../jest.config.js",
    "lint": "eslint --ext .js,.jsx . --ignore-path ../../.eslintignore",
    "precompile": "rm -rf lib",
    "test": "npm run lint && npm run jest"
  }
}
