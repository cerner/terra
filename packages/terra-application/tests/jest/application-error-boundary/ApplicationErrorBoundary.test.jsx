import React from 'react';
import { shallowWithIntl, mountWithIntl } from '@cerner/terra-enzyme-intl';
import ApplicationErrorBoundary from '../../../src/application-error-boundary/ApplicationErrorBoundary';
import Logger from '../../../src/utils/logger';

describe('ApplicationErrorBoundary', () => {
  describe('Snapshots', () => {
    it('should render with minimal props', () => {
      const wrapper = shallowWithIntl((
        <ApplicationErrorBoundary />
      )).dive();

      expect(wrapper).toMatchSnapshot();
    });

    it('should render with children', () => {
      const wrapper = shallowWithIntl((
        <ApplicationErrorBoundary>
          <div>Test child</div>
        </ApplicationErrorBoundary>
      )).dive();

      expect(wrapper).toMatchSnapshot();
    });

    it('should render error view when an error is detected', () => {
      const spy = jest.spyOn(Logger, 'error').mockImplementation(() => {});
      const ErrorComponent = () => <div />;

      const wrapper = mountWithIntl((
        <ApplicationErrorBoundary>
          <ErrorComponent />
        </ApplicationErrorBoundary>
      ));
      expect(wrapper).toMatchSnapshot();

      /**
       * After simulating the error, the error view should be rendered.
       */
      wrapper.find(ErrorComponent).simulateError(new Error('test error'));
      expect(wrapper).toMatchSnapshot();

      /**
       * After rendering the component again, the error view should no longer be rendered.
       */
      wrapper.setProps({});
      expect(wrapper).toMatchSnapshot();
      expect(spy).toHaveBeenCalledTimes(1);
      spy.mockRestore();
    });
  });
});
