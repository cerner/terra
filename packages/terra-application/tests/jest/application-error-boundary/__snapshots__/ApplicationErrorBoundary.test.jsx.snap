// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationErrorBoundary Snapshots should render error view when an error is detected 1`] = `
<injectIntl(ApplicationErrorBoundary)>
  <ApplicationErrorBoundary
    intl={
      Object {
        "$t": [Function],
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "defaultRichTextElements": undefined,
        "fallbackOnEmptyString": true,
        "formatDate": [Function],
        "formatDateTimeRange": [Function],
        "formatDateToParts": [Function],
        "formatDisplayName": [Function],
        "formatList": [Function],
        "formatListToParts": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatNumberToParts": [Function],
        "formatPlural": [Function],
        "formatRelativeTime": [Function],
        "formatTime": [Function],
        "formatTimeToParts": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getDisplayNames": [Function],
          "getListFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralRules": [Function],
          "getRelativeTimeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "onError": [Function],
        "onWarn": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": undefined,
        "wrapRichTextChunksInFragment": undefined,
      }
    }
  >
    <ErrorComponent>
      <div />
    </ErrorComponent>
  </ApplicationErrorBoundary>
</injectIntl(ApplicationErrorBoundary)>
`;

exports[`ApplicationErrorBoundary Snapshots should render error view when an error is detected 2`] = `
<injectIntl(ApplicationErrorBoundary)>
  <ApplicationErrorBoundary
    intl={
      Object {
        "$t": [Function],
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "defaultRichTextElements": undefined,
        "fallbackOnEmptyString": true,
        "formatDate": [Function],
        "formatDateTimeRange": [Function],
        "formatDateToParts": [Function],
        "formatDisplayName": [Function],
        "formatList": [Function],
        "formatListToParts": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatNumberToParts": [Function],
        "formatPlural": [Function],
        "formatRelativeTime": [Function],
        "formatTime": [Function],
        "formatTimeToParts": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getDisplayNames": [Function],
          "getListFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralRules": [Function],
          "getRelativeTimeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "onError": [Function],
        "onWarn": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": undefined,
        "wrapRichTextChunksInFragment": undefined,
      }
    }
  >
    <StatusView
      buttonAttrs={Array []}
      isAlignedTop={false}
      isGlyphHidden={false}
      message="terraApplication.errorBoundary.defaultErrorMessage"
      role="alert"
      variant="error"
    >
      <div
        className="outer-view"
        role="alert"
      >
        <div
          className="top-space"
        />
        <div
          className="inner-view"
        >
          <div
            className="glyph"
          >
            <div
              className="error glyph-icon"
            />
          </div>
          <p
            className="title"
          >
            <FormattedMessage
              id="Terra.status-view.error"
            >
              Terra.status-view.error
            </FormattedMessage>
          </p>
          <div
            className="divider"
          >
            <Divider>
              <hr
                className="divider"
              />
            </Divider>
          </div>
          <p
            className="message"
          >
            terraApplication.errorBoundary.defaultErrorMessage
          </p>
        </div>
        <div
          className="bottom-space"
        />
      </div>
    </StatusView>
  </ApplicationErrorBoundary>
</injectIntl(ApplicationErrorBoundary)>
`;

exports[`ApplicationErrorBoundary Snapshots should render error view when an error is detected 3`] = `
<injectIntl(ApplicationErrorBoundary)>
  <ApplicationErrorBoundary
    intl={
      Object {
        "$t": [Function],
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "defaultRichTextElements": undefined,
        "fallbackOnEmptyString": true,
        "formatDate": [Function],
        "formatDateTimeRange": [Function],
        "formatDateToParts": [Function],
        "formatDisplayName": [Function],
        "formatList": [Function],
        "formatListToParts": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatNumberToParts": [Function],
        "formatPlural": [Function],
        "formatRelativeTime": [Function],
        "formatTime": [Function],
        "formatTimeToParts": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getDisplayNames": [Function],
          "getListFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralRules": [Function],
          "getRelativeTimeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "onError": [Function],
        "onWarn": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": undefined,
        "wrapRichTextChunksInFragment": undefined,
      }
    }
  >
    <ErrorComponent>
      <div />
    </ErrorComponent>
  </ApplicationErrorBoundary>
</injectIntl(ApplicationErrorBoundary)>
`;

exports[`ApplicationErrorBoundary Snapshots should render with children 1`] = `
<ApplicationErrorBoundary
  intl={
    Object {
      "$t": [Function],
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "defaultRichTextElements": undefined,
      "fallbackOnEmptyString": true,
      "formatDate": [Function],
      "formatDateTimeRange": [Function],
      "formatDateToParts": [Function],
      "formatDisplayName": [Function],
      "formatList": [Function],
      "formatListToParts": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatNumberToParts": [Function],
      "formatPlural": [Function],
      "formatRelativeTime": [Function],
      "formatTime": [Function],
      "formatTimeToParts": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getDisplayNames": [Function],
        "getListFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralRules": [Function],
        "getRelativeTimeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "onError": [Function],
      "onWarn": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": undefined,
      "wrapRichTextChunksInFragment": undefined,
    }
  }
>
  <div>
    Test child
  </div>
</ApplicationErrorBoundary>
`;

exports[`ApplicationErrorBoundary Snapshots should render with minimal props 1`] = `
<ApplicationErrorBoundary
  intl={
    Object {
      "$t": [Function],
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "defaultRichTextElements": undefined,
      "fallbackOnEmptyString": true,
      "formatDate": [Function],
      "formatDateTimeRange": [Function],
      "formatDateToParts": [Function],
      "formatDisplayName": [Function],
      "formatList": [Function],
      "formatListToParts": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatNumberToParts": [Function],
      "formatPlural": [Function],
      "formatRelativeTime": [Function],
      "formatTime": [Function],
      "formatTimeToParts": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getDisplayNames": [Function],
        "getListFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralRules": [Function],
        "getRelativeTimeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "onError": [Function],
      "onWarn": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": undefined,
      "wrapRichTextChunksInFragment": undefined,
    }
  }
/>
`;
